### ISSUE TITLE
Resource Template facet doesn't work

### ISSUE THREAD

**Issue by yangli0516** [omeka-s-modules/FacetedBrowse#51]
All other types of facets (value, resource class etc.) work great while the resource template facet is broken. When applying the resource template value in the faceted browse site page, nothing happens, and it always show the full result set.

![image](https://github.com/user-attachments/assets/b5781fc2-6ab0-40d0-9952-64e7f670c1a5)

**Comment by jimsafley** [comment]
I can't reproduce this. The resource template facet works as expected.

Open your browser's web development tools. Are you getting a JS error in the console? When clicking of a template facet, what's the URL of the resulting fetch request? Go to the item advanced search page and see if filtering by template there results in the same behavior.

**Comment by yangli0516** [comment]
@jimsafley there's no JS error in the console. The URL fetching the request when I clicked on "Newspaper" is

`http://omaa-omeka.laragon.local/s/omaa/faceted-browse/1/browse?resource_template_id%5B%5D=2&resource_template_id%5B%5D=3&resource_template_id%5B%5D=4&resource_template_id[]=3&faceted_browse_category_id=1`

Is this caused by setting the search query based on resource templates to the category?

![image](https://github.com/user-attachments/assets/2092f97a-4060-4ad8-9cce-791d28b8f019)

**Comment by yangli0516** [comment]
I have tested removing the search query from the category. Now the resource template facets are working. From my understanding, the search query in a category sets the whole result set for the search, which shouldn't affect the behavior of facets.

**Comment by jimsafley** [comment]
Yes, this is a bug caused by merging the category query with the facets query, which, as you discovered, can conflict with each other. I'm working on a fix that avoids the merge altogether.

**Comment by jimsafley** [comment]
This has been fixed in v1.6.0.
